### ISSUE TITLE
UserWarning: torch.meshgrid

### ISSUE THREAD

**Issue by the-cat-crying** [verlab/accelerated_features#45]
UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  ..\aten\src\ATen\native\TensorShape.cpp:2157.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]

../accelerated_features/third_party/alike_wrapper.py    line 76
../accelerated_features/third_party/ALIKE/soft_detect.py line 88

**Comment by guipotje** [comment]
Hi @the-cat-crying, this is just a warning because the indexing argument will be required in a future release of PyTorch. You can fix this by passing the argument indexing="ij".

**Comment by the-cat-crying** [comment]
Thanks！

**Comment by the-cat-crying** [comment]
But I also found two caveats

1 ../accelerated_features/third_party/ALIKE/soft_detect.py:152: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  keypoints_xy_nms = torch.stack([indices_kpt % w, indices_kpt // w], dim=1)  # Mx2
2 ../accelerated_features/modules/training/losses.py:105: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.
  kpts = F.log_softmax(kpts)

I found that the code in these two places seems to be duplicated, please guide me if this is what I think.

**Comment by the-cat-crying** [comment]
![image](https://github.com/user-attachments/assets/92e90dfc-ffae-4e64-8139-2da2b5cf0cf8)

**Comment by XinningC** [comment]
Hello, I've also encounted "Change the call to include dim=X as an argument." on line "kpts = F.log_softmax(kpts)", and I found out that when the argument dim is not specified, the function takes dim=1 by default, which does not makes trouble. Hope this helps you, and please correctify me if I'm wrong.

**Comment by the-cat-crying** [comment]
Thanks !@XinningC  You are right！
